### ISSUE TITLE
Warning: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD

### ISSUE THREAD

**Issue by Karm** [quarkusio/quarkus#18858]
Quarkus 2.0.2 shows a new Warning while compiling a MicroProfile application (Health checks) with Graal/Mandrel:

```
[WARNING] [io.quarkus.arc.processor.BeanArchives] Failed to index [Lio.smallrye.health.SmallRyeHealthReporter$HealthType;: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD
[WARNING] [io.quarkus.arc.processor.BeanArchives] Failed to index [Ljava.lang.Object;: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD
```

Full log: [gist](https://gist.github.com/Karm/0aaf1317996eb98150881ab76625837c)

### Expected behavior

Explained or fixed.

### How to Reproduce?
(With your GRAALVM_HOME set or Docker available...)

```
git clone https://github.com/Karm/mandrel-integration-tests.git
cd mandrel-integration-tests/apps/quarkus-full-microprofile/
mvn clean compile package -Pnative -Dquarkus.version=2.0.2.Final
```

### Quarkus versions affected

Quarkus 2.0.0.Final was O.K. and Quarkus 2.1.0.CR1 is O.K. again.
Apparently not dependent on Graal/Mandrel version.

![warning](https://user-images.githubusercontent.com/691097/126331209-02c6cee1-a3aa-4e96-b90c-f5d1eabd47d4.png)

### System:

Both Mandrel and Graal distributions:

```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/karm/Tools/apache-maven-3.6.3
Java version: 11.0.12-ea, vendor: Oracle Corporation, runtime: /home/karm/workspaceRH/mandrel-release/mandrel-java11-linux-amd64-20.3.3.1-dev10142b1472/mandrel-java11-20.3-SNAPSHOT
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.18.0-240.22.1.el8_3.x86_64", arch: "amd64", family: "unix"
```

```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/karm/Tools/apache-maven-3.6.3
Java version: 11.0.11, vendor: GraalVM Community, runtime: /var/X/JDKs/graalvm-ce-java11-20.3.2
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.18.0-240.22.1.el8_3.x86_64", arch: "amd64", family: "unix"
```

**Comment by geoand** [comment]
I think this has already been fixed in `2.1.0.CR1` (or `2.1.0.Final`), right @mkouba ?

**Comment by famod** [comment]
The fix will be part of 2.0.3.Final: #18675

**Comment by geoand** [comment]
Great, I'll close this then

**Comment by Karm** [comment]
THX @famod  @geoand :pray:

**Comment by mkouba** [comment]
> I think this has already been fixed in `2.1.0.CR1` (or `2.1.0.Final`), right @mkouba ?

So it was fixed in a way that the bean archive index ignores the `IndexView.getClassByName(DotName)` queries for array-based dot names, i.e. these warnings are not diplayed anymore. Whether it's ok or not depends on the actual use case, i.e. an extension should not attempt to obtain a class for an array type.
